src/transforms/pipelines/config.rs

```diff
@@ -103,35 +103,27 @@ impl PipelineConfig {
 //------------------------------------------------------------------------------
 
 /// This represent an ordered list of pipelines depending on the event type.
-#[derive(Clone, Debug, Default, Deserialize, Serialize)]
-#[serde(deny_unknown_fields)]
-pub(crate) struct EventTypeConfig {
-    #[serde(default)]
-    order: Option<Vec<String>>,
-    pipelines: IndexMap<String, PipelineConfig>,
-}
-
-#[cfg(test)]
-impl EventTypeConfig {
-    #[allow(dead_code)] // for some small subset of feature flags this code is dead
-    pub(crate) const fn order(&self) -> &Option<Vec<String>> {
-        &self.order
-    }
+#[derive(Clone, Debug, Default, Serialize, Deserialize)]
+pub(crate) struct EventTypeConfig(Vec<PipelineConfig>);
 
-    #[allow(dead_code)] // for some small subset of feature flags this code is dead
-    pub(crate) const fn pipelines(&self) -> &IndexMap<String, PipelineConfig> {
-        &self.pipelines
+impl AsRef<Vec<PipelineConfig>> for EventTypeConfig {
+    fn as_ref(&self) -> &Vec<PipelineConfig> {
+        &self.0
     }
 }
 
 impl EventTypeConfig {
+    pub fn is_empty(&self) -> bool {
+        self.0.is_empty()
+    }
+
     pub(super) fn validate_nesting(&self, parents: &HashSet<&'static str>) -> Result<(), String> {
-        for (name, pipeline) in self.pipelines.iter() {
-            for (index, transform) in pipeline.transforms.iter().enumerate() {
+        for (pipeline_index, pipeline) in self.0.iter().enumerate() {
+            for (transform_index, transform) in pipeline.transforms.iter().enumerate() {
                 if !transform.nestable(parents) {
                     return Err(format!(
-                        "the transform {} in pipeline {:?} cannot be nested in {:?}",
-                        index, name, parents
+                        "the transform {} in pipeline {} cannot be nested in {:?}",
+                        transform_index, pipeline_index, parents
```

Comment:
Could we maybe pull the `name` field of the pipeline and the `id` of the transform? I think that would make it easier for users to find the offending transform.